### PR TITLE
35672435: Added a customized tldraw whiteboard with numerous features, including custom tools, zoom functionality, fullscreen mode, and more.

### PR DESCRIPTION
**Pull request by Leyofficial**
### A significant amount of work has been done with the tldraw js library. This includes working on all the tools in the toolbar to ensure a maximum combination of functionality and originality, as implemented in the design. Considerable effort was put into the canvas setup and full scalability, as well as the movement of elements already present on the board. All hotkeys for quick zoom and tool management are in place and fully operational. All icons were added to xipkg/icons, which are also present on the npm site. Additionally, extensive work was done on the board structure and how it will be customized in the future. This was an important part as readability, ease, and optimized code were key goals.

🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉

### **Examples :** 

Clean board
Снимок экрана 2024-07-22 в 16 45 36

Board with a sticker, text and pen.
Снимок экрана 2024-07-22 в 16 46 43

Full screen mode
Снимок экрана 2024-07-22 в 16 47 19

Zoom out
Снимок экрана 2024-07-22 в 16 47 46

Zoom in
Снимок экрана 2024-07-22 в 16 48 14

Undo button
Снимок экрана 2024-07-22 в 16 48 35

Redo button
Снимок экрана 2024-07-22 в 16 48 50

Arrow tool
Снимок экрана 2024-07-22 в 16 51 08

Eraser tool
Снимок экрана 2024-07-22 в 16 52 13

### CODE REVIEW

File: packages/pkg.module.whiteboard/components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Minimize, Maximize } from '@xipkg/icons';
+import { useFullScreen } from '../../pkg.utils.client/useFullScreen';
+export function Header() {
+  const { isFullScreen, toggleFullScreen } = useFullScreen('whiteboard-container');
+  return (
+    <div className="absolute z-[300] w-full bg-white px-4 py-2 text-3xl">
```
Comment (unknownproperty) [inline packages/pkg.module.whiteboard/components/Header.tsx]: `bg-gray-0`

File: packages/pkg.module.whiteboard/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      <div className="flex items-center justify-between">
+        <div className="flex gap-2">
+          <h1 className="text-[24px]">B1.2</h1>
```
Comment (unknownproperty) [inline packages/pkg.module.whiteboard/components/Header.tsx]: Используй, пожалуйста, для текста свойства из нашего конфига tailwind из последнего сообщения в объявления-ф

Так для всего текста

File: packages/pkg.module.whiteboard/components/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+          <p className="text-gray-60 py-1 text-[16px]">Intermediate</p>
+        </div>
+        <div className="flex items-center gap-2">
+          <button onClick={toggleFullScreen} type="button" className="bg-transparent fill-gray-100">
```
Comment (unknownproperty) [inline packages/pkg.module.whiteboard/components/Header.tsx:13]: А давай лучше использовать Button variant="ghost"

File: packages/pkg.module.whiteboard/components/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+type TNavbarElement = {
+  action: string;
+  title: string;
+  icon: React.ReactNode | null;
+};
+
+export const Navbar = track(() => {
+  const editor = useEditor();
+
+  const navBarElements: TNavbarElement[] = [
```
Comment (unknownproperty) [inline packages/pkg.module.whiteboard/components/Navbar.tsx]: Нууу, как будто всё же константы мы выносим за пределы компонент

File: packages/pkg.module.whiteboard/components/Navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    { action: 'arrow', title: 'Arrow', icon: <Arrow /> },
+    { action: 'image', title: 'Image', icon: <Image /> },
+    { action: 'eraser', title: 'Eraser', icon: <Eraser /> },
+  ];
+
+  return (
+    <div className="pointer-events-none absolute inset-0">
+      <div className="absolute bottom-3 left-0 right-0 z-[300] flex w-full items-center justify-center">
+        <div className="z-[300] flex gap-7">
+          <div className="border-gray-10 z-[300] rounded-[12px] border bg-white p-2">
```
Comment (unknownproperty) [inline packages/pkg.module.whiteboard/components/Navbar.tsx]: `bg-gray-0`

File: packages/pkg.module.whiteboard/components/Navbar.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            <NavbarAction />
+          </div>
+          <div className="border-gray-10 flex gap-10 rounded-[12px] border bg-white">
```
Comment (unknownproperty) [inline packages/pkg.module.whiteboard/components/Navbar.tsx]: `bg-gray-0`

File: packages/pkg.module.whiteboard/components/Navbar.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            <div className="flex gap-2 p-1">
+              {navBarElements.map((item: TNavbarElement) => (
+                <button
+                  key={item.action}
+                  className={`pointer-events-auto flex h-[32px] w-[32px] items-center justify-center rounded-[8px] ${editor.getCurrentToolId() == item.action ? 'bg-brand-0' : 'bg-white'}`}
```
Comment (unknownproperty) [inline packages/pkg.module.whiteboard/components/Navbar.tsx]: `bg-gray-0`

File: packages/pkg.module.whiteboard/components/NavbarAction.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import { Undo, Redo } from '@xipkg/icons';
+
+export function NavbarAction() {
+  const editor = useEditor();
+  const canRedo = useCanRedo();
+  const canUndo = useCanUndo();
+
+  return (
+    <div className="pointer-events-auto flex items-center justify-center gap-5">
+      <button className="items-center rounded-[8px] bg-white" onClick={() => editor.undo()}>
```
Comment (unknownproperty) [inline packages/pkg.module.whiteboard/components/NavbarAction.tsx]: `bg-gray-0`

File: packages/pkg.module.whiteboard/components/NavbarAction.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+        <Undo className={`${canUndo ? null : 'fill-gray-40'}`} />
+      </button>
+      <button className="items-center rounded-[8px] bg-white" onClick={() => editor.redo()}>
```
Comment (unknownproperty) [inline packages/pkg.module.whiteboard/components/NavbarAction.tsx]: `bg-gray-0`

File: packages/pkg.module.whiteboard/components/ZoomMenu.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+  const handleZoomIn = () =>
+    editor.zoomIn(editor.getViewportScreenCenter(), { animation: { duration: 200 } });
+
+  const handleZoomOut = () =>
+    editor.zoomOut(editor.getViewportScreenCenter(), { animation: { duration: 120 } });
+
+  return (
+    <div className="absolute bottom-3 right-3 z-[300]">
+      <div className="rounded-[16px] bg-white">
```
Comment (unknownproperty) [inline packages/pkg.module.whiteboard/components/ZoomMenu.tsx]: `bg-gray-0`

**Comment by Leyofficial** [comment]
### Fixed all comments in Pull Request.

**Review by unknownproperty — APPROVED** [review]

**Pull request merged**